Makefile

```diff
@@ -2,26 +2,50 @@ HOSTNAME=github.com
 NAMESPACE=iterative
 NAME=iterative
 VERSION=0.0.${shell date +%s}+development
-OS_ARCH=${shell go env GOOS}_${shell go env GOARCH}
-BINARY=terraform-provider-${NAME}
-INSTALL_PATH=~/.terraform.d/plugins/${HOSTNAME}/${NAMESPACE}/${NAME}/${VERSION}/${OS_ARCH}
+GOOS=${shell go env GOOS}
+GOARCH=${shell go env GOARCH}
+TF_PLUGIN_INSTALL_PATH=~/.terraform.d/plugins/${HOSTNAME}/${NAMESPACE}/${NAME}/${VERSION}/${GOOS}_${GOARCH}
+TPI_PATH ?= $(shell pwd)
+LEO_PATH ?= $(shell pwd)/cmd/leo
+GO_LINK_FLAGS ?= -s -w -X terraform-provider-iterative/iterative/utils.Version=${VERSION}
+
+LEO_BUILD_COMMAND ?= \
+	CGO_ENABLED=0 \
+	go build -ldflags="$(GO_LINK_FLAGS)" \
+	-o $(shell pwd)/leo-$(GOOS)-$(GOARCH) \
+	$(LEO_PATH)
```

Comment:
Worth moving to `leo-bin` target instead of keeping a separate variable?

Reply:
sure 👍 No strong opinion  - when/if we have more commands it might make sense to group them on top and not burry them in the targets)